src/withAxiom.ts

```diff
@@ -101,14 +102,40 @@ function withAxiomNextApiHandler(handler: NextApiHandler): NextApiHandler {
   };
 }
 
+export type AxiomRequest = NextRequest & { log: Logger };
+export type AxiomMiddleware = (
+  request: AxiomRequest,
+  event: NextFetchEvent
+) => NextMiddlewareResult | Promise<NextMiddlewareResult>;
```

Comment:
I think we'll also need an `AxiomApiRequest`, the API function logs have the same issue don't they?

Reply:
on it.

Reply:
the NextAPIRequest structure is different than NextRequest. I can't find request ip and geo info for example.

Reply:
Are you still blocked on this @schehata ?

Reply:
I was able to get the IP from the headers, and for the region information I used VERCEL_REGION. I believe we can move forward with those